main.go

```diff
@@ -0,0 +1,517 @@
+					if Type_IsArray(t) {
+						WriteStructHeader(context, f.Names[0].Name, TD_STRUCT_ARRAY)
+						WriteFieldStruct(context, f)
+					} else {
+						WriteStructHeader(context, f.Names[0].Name, TD_STRUCT)
+						WriteFieldStruct(context, f)
+					}
+				}
+			}
+		}
+	} else {
+		// `toml` requires that "anonymous" fields be structs. Otherwise,
+		// the field is ignored completely. `toml` requires all fields to
+		// be exported. "anonymous" fields are not exported. However, a
+		// struct may contain an exported "named" field.
+		if Type_IsStruct(t) && (flags&TD_FOLLOW != 0) {
+			WriteFieldStruct(context.DecIndent(), f)
+		}
+	}
+
+}
+
+func ProcessStruct(context *Context, s_name string) {
+	for _, f := range context.Package.Files {
+		ast.Inspect(f, func(node ast.Node) bool {
+			decl, ok := node.(*ast.GenDecl)
+			if !ok {
+				return true
+			}
+
+			if len(decl.Specs) < 1 {
+				return false
+			}
+
+			spec, ok := decl.Specs[0].(*ast.TypeSpec)
+			if !ok {
+				return false
+			}
+
+			s, ok := spec.Type.(*ast.StructType)
+			if !ok {
+				return false
+			}
+
+			if spec.Name.Name != s_name {
+				return false
+			}
+
+			if decl.Doc != nil {
+				WriteComment(context, decl.Doc)
+			}
+			for _, f := range s.Fields.List {
+				ProcessField(context, f)
+			}
+
+			return true
+		})
+	}
+}
+
+func main() {
+	var ft = flag.String("t", "", "(Required) Specifies the target structure.")
+	var fo = flag.String("o", "",
+		`This specifies the output path for the generated toml document.
+It is recommended that the location is in the same directory or
+a subdirectory of the current package. This option is
+required.`)
+	var fp = flag.String("p", "",
+		`This specifies the target package. This is only required when
+"$GOPACKAGE" is not present in the environment. Therefore, this
+can be ignored when used as a "go generate" directive. If this
+is supplied while "$GOPACKAGE" is present, this takes priority.`)
+	flag.Parse()
+
+	ep := os.Getenv("GOPACKAGE")
+
+	// Ensures that the "target" and "output" parameters are always
+	// specified. This conditionally ensures that either the
+	// parameters or the environment specify the "package"
+	// parameter.
+	if (*ft == "" || *fo == "") ||
+		(*fp == "" && ep == "") {
+		flag.Usage()
+        os.Exit(1)
+	}
+
+	p_target := ep
+	if *fp != "" {
+		p_target = *fp
+	}
+
+	// Attempt to load information about packages from the source code
+	// located in the current working directory. This does not include
+	// information from external packages. If an external package is
+	// required to build metadata for the target struct, it should
+	// be download and available in GOROOT.
+	found, err := packages.Load(
+		&packages.Config{
+			Mode: packages.NeedName |
+				packages.NeedTypes |
+				packages.NeedTypesInfo |
+				packages.NeedSyntax |
+				packages.NeedImports |
+				packages.NeedFiles,
+			Tests: false,
+		},
+	)
+	if err != nil {
+		panic(err)
+	}
+
+	f, err := os.Create(*fo)
```

Comment:
What happens here is the `o` flag is not specified?

Reply:
There's a prior check that ensures that `fo` isn't an empty string (flags initializes strings to "" when nothing is specified for that argument). If `fo` is an empty string, it prints the help text and exits. If we need to, I can add an additional check here for safety.